pkg/controller/resources/route.go

```diff
@@ -25,12 +25,22 @@ const (
 
 var (
 	// ErrNotSupportedTLSTermination is an error when unsupported TLS termination is configured via annotation.
-	ErrNotSupportedTLSTermination = errors.New("Not supported tls termination is specified. Only \"passthrough\" is valid.")
+	ErrNotSupportedTLSTermination = errors.New("not supported tls termination is specified. Only \"passthrough\" is valid")
+
+	// ErrNoValidLoadbalancerDomain indicates that the current ingress does not have a DomainInternal field, or
+	// said field does not contain a value we can work with.
+	ErrNoValidLoadbalancerDomain = errors.New("unable to find ClusterIngress LoadBalancer with DomainInternal set")
 )
 
 // MakeRoutes creates OpenShift Routes from a Knative Ingress
 func MakeRoutes(ci networkingv1alpha1.IngressAccessor) ([]*routev1.Route, error) {
 	routes := []*routev1.Route{}
+
+	// Skip all route creation for cluster-local ingresses.
+	if ci.GetSpec().Visibility == networkingv1alpha1.IngressVisibilityClusterLocal {
+		return routes, nil
+	}
```

Comment:
I see `MakeRoutes` has been called only when 
https://github.com/openshift-knative/knative-openshift-ingress/blob/master/pkg/controller/common/reconciler.go#L31

So is there any chance that `Visibility` having clusterLocal ?

Reply:
Probably not in this context. I'm leaning towards keeping it anyway for self-sufficient correctness of this function.